pkg/tfbridge/provider.go

```diff
@@ -351,10 +351,18 @@ func buildTerraformConfig(p *Provider, vars resource.PropertyMap) (shim.Resource
 		}
 	}
 
-	// Make a Terraform config map out of the variables. We do this before checking for missing properties
-	// s.t. we can pull any defaults out of the TF schema.
-	config, _, err := MakeTerraformConfig(p, tfVars, p.config, p.info.Config)
-	return config, err
+	ctx := conversionContext{
+		ProviderConfig: p.configValues,
+		Assets:         AssetTable{},
+		ApplyDefaults:  true,
+	}
+
+	inputs, err := ctx.MakeTerraformInputs(nil, tfVars, p.config, p.info.Config, false)
```

Comment:
You should be able to just call the top-level `MakeTerraformInputs` function, which will take care of creating the `conversionContext`.

Reply:
Nice! updated and tested that it works as expected with the Keycloak and AzureAD providers (where I had issues with this)